### PR TITLE
8283695: [AIX] Build failure due to name conflict in test_arguments.cpp

### PR DESCRIPTION
**Pull request by backwaterred**
The recent addition of a variable named 'BAD' in test_arguments.cpp clashes with a macro of the same name in 'sys/param.h' on AIX. Renaming the variable is sufficient to fix this build issue.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283695](https://bugs.openjdk.java.net/browse/JDK-8283695): [AIX] Build failure due to name conflict in test_arguments.cpp

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7962/head:pull/7962` \
`$ git checkout pull/7962`

Update a local copy of the PR: \
`$ git checkout pull/7962` \
`$ git pull https://git.openjdk.java.net/jdk pull/7962/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7962`

View PR using the GUI difftool: \
`$ git pr show -t 7962`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7962.diff">https://git.openjdk.java.net/jdk/pull/7962.diff</a>

</details>

### CODE REVIEW

**Comment by iklam** [comment]
The changes look good to me, but the bug title and description should use test_arguments.cpp instead of test_properties.cpp :-)

File: test/hotspot/gtest/runtime/test_arguments.cpp [hunk 1]
```diff
@@ -292,143 +292,143 @@ void check_numeric_flag(JVMFlag* flag, T getvalue(JVMFlag* flag),
 #define INTEGER_TEST_TABLE(f) \
   /*input                      i32           u32           i64                      u64 */ \
   f("0",                       0,            0,            0,                       0                        ) \
-  f("-0",                      0,            BAD,          0,                       BAD                      ) \
-  f("-1",                     -1,            BAD,         -1,                       BAD                      ) \
+  f("-0",                      0,            BAD_INT,      0,                       BAD_INT                      ) \
```
Comment (iklam) [inline test/hotspot/gtest/runtime/test_arguments.cpp:295]: The close parenthesis should be aligned as well.

**Comment by backwaterred** [comment]
I believe the TestLinkPlatform.java failure is unrelated to this change.

**Review by iklam — APPROVED** [review]
LGTM

**Review by tstuefe — APPROVED** [review]
All good. 

IBM has this bad habit of naming things in their headers without considering potential clashes. I remember the define "IA64" - in AIX headers! - giving us headaches in our itanium port.

**Pull request closed without merging**